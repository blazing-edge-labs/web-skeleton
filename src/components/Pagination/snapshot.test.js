import React from 'react';
import renderer from 'react-test-renderer';
import Pagination from './';

describe('Pagination component snapshot', () => {
  it('renders data with current page 1', () => {
    const tree = renderer.create(
      <Pagination
        currentPage={1}
        itemsPerPage={10}
        onPaginationChange={jest.fn()}
        total={127}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders data with current page 5', () => {
    const tree = renderer.create(
      <Pagination
        currentPage={5}
        itemsPerPage={10}
        onPaginationChange={jest.fn()}
        total={93}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders data with current page being last', () => {
    const tree = renderer.create(
      <Pagination
        currentPage={8}
        itemsPerPage={10}
        onPaginationChange={jest.fn()}
        total={78}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders data with less than 5 pages', () => {
    const tree = renderer.create(
      <Pagination
        currentPage={2}
        itemsPerPage={10}
        onPaginationChange={jest.fn()}
        total={30}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders just message for small number of items', () => {
    const tree = renderer.create(
      <Pagination
        currentPage={1}
        itemsPerPage={10}
        onPaginationChange={jest.fn()}
        total={5}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders just message for 1 item', () => {
    const tree = renderer.create(
      <Pagination
        currentPage={1}
        itemsPerPage={10}
        onPaginationChange={jest.fn()}
        total={1}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('doesn\'t render anything for 0 items', () => {
    const tree = renderer.create(
      <Pagination
        currentPage={1}
        itemsPerPage={10}
        onPaginationChange={jest.fn()}
        total={0}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
