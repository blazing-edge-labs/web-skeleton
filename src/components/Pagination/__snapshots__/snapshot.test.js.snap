exports[`Pagination component snapshot doesn't render anything for 0 items 1`] = `null`;

exports[`Pagination component snapshot renders data with current page 1 1`] = `
<div>
  <p>
    Showing 1-10 of 127 results
  </p>
  <ul>
    <li>
      1
    </li>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        2
      </button>
    </li>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        3
      </button>
    </li>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        4
      </button>
    </li>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        5
      </button>
    </li>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        <i
          className="fa fa-angle-right" />
      </button>
    </li>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        <i
          className="fa fa-angle-double-right" />
      </button>
    </li>
  </ul>
</div>
`;

exports[`Pagination component snapshot renders data with current page 5 1`] = `
<div>
  <p>
    Showing 41-50 of 93 results
  </p>
  <ul>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        <i
          className="fa fa-angle-double-left" />
      </button>
    </li>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        <i
          className="fa fa-angle-left" />
      </button>
    </li>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        3
      </button>
    </li>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        4
      </button>
    </li>
    <li>
      5
    </li>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        6
      </button>
    </li>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        7
      </button>
    </li>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        <i
          className="fa fa-angle-right" />
      </button>
    </li>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        <i
          className="fa fa-angle-double-right" />
      </button>
    </li>
  </ul>
</div>
`;

exports[`Pagination component snapshot renders data with current page being last 1`] = `
<div>
  <p>
    Showing 71-78 of 78 results
  </p>
  <ul>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        <i
          className="fa fa-angle-double-left" />
      </button>
    </li>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        <i
          className="fa fa-angle-left" />
      </button>
    </li>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        4
      </button>
    </li>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        5
      </button>
    </li>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        6
      </button>
    </li>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        7
      </button>
    </li>
    <li>
      8
    </li>
  </ul>
</div>
`;

exports[`Pagination component snapshot renders data with less than 5 pages 1`] = `
<div>
  <p>
    Showing 11-20 of 30 results
  </p>
  <ul>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        <i
          className="fa fa-angle-left" />
      </button>
    </li>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        1
      </button>
    </li>
    <li>
      2
    </li>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        3
      </button>
    </li>
    <li>
      <button
        className="empty_6ajk08"
        onClick={[Function]}
        type="button">
        <i
          className="fa fa-angle-right" />
      </button>
    </li>
  </ul>
</div>
`;

exports[`Pagination component snapshot renders just message for 1 item 1`] = `
<div>
  <p>
    Showing 1 result
  </p>
</div>
`;

exports[`Pagination component snapshot renders just message for small number of items 1`] = `
<div>
  <p>
    Showing all of 5 results
  </p>
</div>
`;
